statsmodels/regression/_prediction.py

```diff
@@ -156,20 +156,39 @@ def get_prediction(self, exog=None, transform=True, weights=None,
            (weights.ndim != 1 or weights.shape[0] == exog.shape[1])):
             raise ValueError('weights has wrong shape')
 
+    else:
+        # this handles the case where we don't want to apply weights
+        weights = 1.
     ### end
 
     if pred_kwds is None:
         pred_kwds = {}
     predicted_mean = self.model.predict(self.params, exog, **pred_kwds)
 
-    covb = self.cov_params()
-    var_pred_mean = (exog * np.dot(covb, exog.T).T).sum(1)
+    # handle var_pred_mean for the PHReg case
+    if 'pred_type' in pred_kwds:
+        cov_params = pred_kwds['cov_params']
```

Comment:
What is the usecase for overriding the attached default `cov_params`?
Is this useful in general, so it should be added as a main keyword option and not just for `lhr`?

(we allow it in t_test and wald_test, but not in discrete get_margeff margins.)
The old keyword in t_test uses a shorter keyword  `cov_p=None`, I guess from a time when having shorter names was still a stronger consideration.)